File: app/src/main/java/com/bitflaker/lucidsourcekit/general/database/values/DreamTypes.java
package com.bitflaker.lucidsourcekit.general.database.values;

public enum DreamTypes {
    Nightmare("NTM"),
    SleepParalysis("SPL"),
    FalseAwakening("FAW"),
    Lucid("LCD"),
    Recurring("REC"),
    None("");

    private String id;

    DreamTypes(String id) {
        this.id = id;
    }

    public String getId() {
        return id;
    }

    public static DreamTypes getEnum(String id){
        for (DreamTypes enm : DreamTypes.values()) {
            if (enm.getId().equalsIgnoreCase(id)) {
                return enm;
            }
        }
        return null;
    }
}

File: app/src/main/java/com/bitflaker/lucidsourcekit/main/dreamjournal/rating/DreamTypes.kt
package com.bitflaker.lucidsourcekit.main.dreamjournal.rating

enum class DreamTypes(val id: String) {
    Nightmare("NTM"),
    SleepParalysis("SPL"),
    FalseAwakening("FAW"),
    Lucid("LCD"),
    Recurring("REC"),
    None("");

    companion object {
        @JvmStatic
        fun getEnum(id: String) = DreamTypes.entries.find {
            it.id.equals(id, ignoreCase = true)
        }
    }
}
